server/image_list_libpod.go

```diff
@@ -0,0 +1,77 @@
+// +build libpod
+
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/containers/storage"
+	"github.com/projectatomic/libpod/libpod/image"
+	"github.com/sirupsen/logrus"
+	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+// ListImages lists existing images.
+func (s *Server) ListImages(ctx context.Context, req *pb.ListImagesRequest) (resp *pb.ListImagesResponse, err error) {
+	const operation = "list_images"
+	defer func() {
+		recordOperation(operation, time.Now())
+		recordError(operation, err)
+	}()
+
+	logrus.Debugf("ListImagesRequest: %+v", req)
+	filter := ""
+	reqFilter := req.GetFilter()
+	if reqFilter != nil {
+		filterImage := reqFilter.GetImage()
+		if filterImage != nil {
+			filter = filterImage.Image
+		}
+	}
+	options := storage.StoreOptions{
+		RunRoot:   s.Config().RunRoot,
+		GraphRoot: s.Config().RootConfig.Root,
+	}
+
+	// TODO fix the filtering.
+	_ = filter
+	r, err := image.NewImageRuntimeFromOptions(options)
+
+	if err != nil {
+		return nil, err
+	}
+
+	results, err := r.GetImages()
+	if err != nil {
+		return nil, err
+	}
+
+	resp = &pb.ListImagesResponse{}
+	for _, result := range results {
+		resImg := &pb.Image{
+			Id:          result.ID(),
+			RepoTags:    result.Names(),
+			RepoDigests: result.RepoDigests(),
+		}
+		// TODO libpod image does not expose the USER element yet.  When it does, we
+		// need to re-enable this.
+		uid, username := getUserFromImage(result.User)
+		if uid != nil {
+			resImg.Uid = &pb.Int64Value{Value: *uid}
+		}
+		resImg.Username = username
+		size, err := result.Size(getContext())
```

Comment:
`ListImages` gets a `Context` as an argument, so you can use `ctx` here instead of calling `getContext()`.

Reply:
fixed